components/resolved_ts/src/resolver.rs

```diff
@@ -541,15 +544,21 @@ impl Resolver {
         }
     }
 
+    // This may be inaccurate for large transactions. But it's just for monitoring
+    // and diagnosis.
+    // The inaccuracy comes from untracking large txn locks, because we do not know
```

Comment:
Tracking same key in different generations can also lead to inaccuracy, the same lock will be calculated multi times. Maybe also comment this case.

Reply:
Good catch